### PR TITLE
texlab 1.8.0 (new formula)

### PR DESCRIPTION
**Pull request by clason**
Implementation of the Language Server Protocol for LaTeX
homepage https://texlab.netlify.com/

- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

(new try for #46989, now without node build-time dependency)

### CODE REVIEW

File: Formula/texlab.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  sha256 "af644d2555c3852513135e87dc6f9bc8b5ee789a4f1c151f4478d108fa007c49"
+  head "https://github.com/latex-lsp/texlab.git"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", "--locked",
+                               "--root", prefix,
+                               "--path", "."
+    bin.install "target/release/texlab"
```
Comment (dawidd6) [inline Formula/texlab.rb]: Is this line necessary? Shouldn't `cargo install` put the binary in place already?

File: Formula/texlab.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  desc "Implementation of the Language Server Protocol for LaTeX"
+  homepage "https://texlab.netlify.com/"
+  url "https://github.com/latex-lsp/texlab/archive/v1.8.0.tar.gz"
+  sha256 "af644d2555c3852513135e87dc6f9bc8b5ee789a4f1c151f4478d108fa007c49"
+  head "https://github.com/latex-lsp/texlab.git"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", "--locked",
```
Comment (dawidd6) [inline Formula/texlab.rb]: This can be a single line, no need to split it I think.

Comment (clason) [reply Formula/texlab.rb]: I followed other formulas, but I can change it of course.

File: Formula/texlab.rb (hunk 1 above)
Comment (clason) [reply Formula/texlab.rb]: Apparently not, I'll remove it.

**Review by chenrui333 — APPROVED** [review]
👍

**Comment by chenrui333** [comment]
Thanks @clason for your contribution!!

**Pull request closed without merging**